pandas/tools/tile.py

```diff
@@ -258,12 +283,16 @@ def _format_levels(bins, prec, right=True,
     else:
         levels = ['[%s, %s)' % (fmt(a), fmt(b))
                   for a, b in zip(bins, bins[1:])]
-
     return levels
 
 
-def _format_label(x, precision=3):
+def _format_label(x, precision=3, dtype=None):
     fmt_str = '%%.%dg' % precision
+
+    if dtype == np.datetime64:
```

Comment:
Shouldn't expensive. Alternative option is to use temporary variable. Including other answers, following impl is on my mind.

- https://github.com/pandas-dev/pandas/blob/master/pandas/core/algorithms.py#L400